packages/graph-engine/src/graph/graph.ts

```diff
@@ -229,34 +229,119 @@ export class Graph {
 		sourceHandle: Output,
 		target: Node,
 		targetHandle: Input,
-		variadicIndex: number = -1
+		variadicIndex = -1
 	): Edge {
-		//If its variadic we need to check the existing edges
-		let annotations = {};
 		if (targetHandle.variadic) {
-			const edges = this.inEdges(target.id, targetHandle.name);
-			//The number of edges is the new index
-			annotations = {
-				[annotatedVariadicIndex]:
-					variadicIndex == -1 ? edges.length : variadicIndex
-			} as VariadicEdgeData;
-		}
-		//Check to see if there is already a connection on the target
-		if (targetHandle._edges.length > 0 && !targetHandle.variadic) {
-			throw new Error(
-				`Input ${targetHandle.name} on node ${target.id} is already connected`
-			);
-		}
+			// negative => error (except -1 which means append)
+			if (variadicIndex < -1) {
+				throw new Error('Invalid variadic index');
+			}
+			// existing edges
+			const existingEdges = this.inEdges(target.id, targetHandle.name);
+			// sort
+			existingEdges.sort((a, b) => {
+				const av = a.annotations[annotatedVariadicIndex] ?? 0;
+				const bv = b.annotations[annotatedVariadicIndex] ?? 0;
+				return av - bv;
+			});
+			// if -1 => append
+			if (variadicIndex === -1) {
+				variadicIndex = existingEdges.length;
+			}
+			// if larger => fill placeholders
+			while (existingEdges.length < variadicIndex) {
```

Comment:
No, lets not do this . Spamming placeholders is not a good idea here. If the issue is that the variadic is causing a problem on load we should rather handle that in the deserialization phase. This will just complicate the system and likely cause more issues